internal/resourcepolicies/resource_policies.go

```diff
@@ -122,16 +135,27 @@ func (p *Policies) match(res *structuredVolume) *Action {
 	return nil
 }
 
-func (p *Policies) GetMatchAction(res any) (*Action, error) {
+func (p *Policies) GetMatchAction(volumeRes any, pvcRes any) (*Action, error) {
 	volume := &structuredVolume{}
-	switch obj := res.(type) {
+	switch obj := volumeRes.(type) {
 	case *v1.PersistentVolume:
 		volume.parsePV(obj)
 	case *v1.Volume:
 		volume.parsePodVolume(obj)
 	default:
 		return nil, errors.New("failed to convert object")
 	}
+
```

Comment:
This block should probably be within the `case *v1.PersistentVolume` block, since secondaryRes as PVC is only expected if res is a PV.